### PR TITLE
Resolve linter issues

### PR DESCRIPTION
**Pull request by shubham1172**
# Description

Per #184, we stopped redirecting `npm run lint` to `/dev/null`. This PR fixes some existing linting issues. 

## Issue reference

Please reference the issue this PR will close: #188 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [X] Code compiles correctly
* [ ] Created/updated tests (NA
* [ ] Extended the documentation (NA)

### CODE REVIEW

**Comment by shubham1172** [comment]
@XavierGeerinck, should we enforce linting, going forward?

File: examples/configuration/src/index.ts [hunk 1]
```diff
@@ -1,13 +1,13 @@
-import { DaprClient, CommunicationProtocolEnum } from "dapr-client";
+import { DaprClient } from "dapr-client";
 
 const daprHost = "127.0.0.1";
-const daprAppId = "example-config";
+const daprPortDefault = "30000";
```
Comment (shubham1172) [inline examples/configuration/src/index.ts:4]: Is there another sane default for this?

File: test/unit/actor/actorRuntime.test.ts [hunk 2]
```diff
@@ -115,10 +114,10 @@ describe('ActorRuntime', () => {
   });
 
   it('should be able to fire a reminder', async () => {
-    const actorId = new ActorId(uuidv4());
+    new ActorId(uuidv4());
   });
 
   it('should be able to fire a timer', async () => {
-    const actorId = new ActorId(uuidv4());
+    new ActorId(uuidv4());
```
Comment (shubham1172) [inline test/unit/actor/actorRuntime.test.ts:121]: Both these tests for `reminder` and `timer` don't seem to be doing anything, am I missing something or can this be cleaned up/marked as todo?

File: test/e2e/main.grpc.test.ts [hunk 3]
```diff
@@ -34,7 +34,7 @@ describe('grpc/main', () => {
 
   describe('metadata', () => {
     it('should be able to get the metadata of the Dapr sidecar', async () => {
-      const res = await client.metadata.get();
+      await client.metadata.get();
```
Comment (shubham1172) [inline test/e2e/main.grpc.test.ts:37]: This test is also commented out. Should be clean this up or mark this as todo?

File: src/implementation/Client/HTTPClient/health.ts [hunk 4]
```diff
@@ -12,7 +12,7 @@ export default class HTTPClientHealth implements IClientHealth {
   // Send an event to an external system
   async isHealthy(): Promise<boolean> {
     try {
-      const result = await this.client.execute(`/healthz`, {
+      await this.client.execute(`/healthz`, {
```
Comment (shubham1172) [inline src/implementation/Client/HTTPClient/health.ts:15]: Shouldn't we be checking result's status code for returning?

File: examples/configuration/src/index.ts (hunk 1 above)
Comment (XavierGeerinck) [reply examples/configuration/src/index.ts:4]: 3500 is the default http port for dapr

File: src/actors/runtime/AbstractActor.ts [hunk 5]
```diff
@@ -71,7 +71,7 @@ export default abstract class AbstractActor {
    * @param <Type> Type of the state object
    * @return Async void response
    */
-  async registerActorReminder<Type>(reminderName: string, dueTime: Temporal.Duration, period: Temporal.Duration, state?: any) {
+  async registerActorReminder<_Type>(reminderName: string, dueTime: Temporal.Duration, period: Temporal.Duration, state?: any) {
```
Comment (XavierGeerinck) [inline src/actors/runtime/AbstractActor.ts:74]: Why are we using underscores?

File: test/unit/actor/actorRuntime.test.ts (hunk 2 above)
Comment (XavierGeerinck) [reply test/unit/actor/actorRuntime.test.ts:121]: The unit tests are not there yet for this, so we should mark them as todo

File: src/implementation/Client/HTTPClient/health.ts (hunk 4 above)
Comment (shubham1172) [reply src/implementation/Client/HTTPClient/health.ts:15]: Got it!

File: src/actors/runtime/AbstractActor.ts (hunk 5 above)
Comment (shubham1172) [reply src/actors/runtime/AbstractActor.ts:74]: _XYZ is a general convention for args/params that are unused. I have added the same to eslintrc file too.

Comment (XavierGeerinck) [reply src/actors/runtime/AbstractActor.ts:74]: Could you go a bit more in depth on which linting rule this solves? It's a bit new for me the underscore usage. As far as I could find it is related to Python a lot, and when I research it for Typescript, they often have a "Ban leading underscore" rule. So I am unsure this is the way to go personally

Comment (shubham1172) [reply src/actors/runtime/AbstractActor.ts:74]: The following in `.eslintrc.json` ignores that.
```json
        "@typescript-eslint/no-unused-vars": ["warn", {
            "varsIgnorePattern": "^_",
            "argsIgnorePattern": "^_"
        }]
```
As far as where the standard comes from comes, I saw it at multiple places. Example, https://github.com/microsoft/TypeScript/issues/9458  https://stackoverflow.com/questions/38835001/skip-type-check-on-unused-parameters

Comment (XavierGeerinck) [reply src/actors/runtime/AbstractActor.ts:74]: Ok, thanks for clarifying!

**Comment by XavierGeerinck** [comment]
LGTM

**Pull request merged**